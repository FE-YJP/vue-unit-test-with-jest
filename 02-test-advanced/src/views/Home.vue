<template>
  <div class="home">
    <v-row align="center">
      <v-col class="text-center">
        <div class="my-2 ">
          <v-btn to="test" large depressed  color="error">点我跳转</v-btn>
        </div>
      </v-col>
    </v-row>
    <!-- <img alt="Vue logo" src="../assets/logo.png"> -->
    <HelloWorld msg="Welcome to Your Vue.js App"/>
  </div>
</template>

<script>
// @ is an alias to /src
import HelloWorld from '@/components/HelloWorld/HelloWorld.vue'

export default {
  name: 'Home',
  components: {
    HelloWorld
  }
}
</script>
