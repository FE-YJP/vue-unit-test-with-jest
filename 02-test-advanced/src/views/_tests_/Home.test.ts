import { shallowMount } from '@vue/test-utils'
import Home from '../Home.vue'

describe('views.Home', () => {
  describe('snapshot', () => {
    it('snapshot测试', () => {
      const wrapper = shallowMount(Home)
      expect(wrapper.vm.$el).toMatchSnapshot()
    })
  })
})
