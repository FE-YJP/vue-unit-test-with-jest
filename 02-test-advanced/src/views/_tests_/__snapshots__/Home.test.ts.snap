// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views.Home snapshot snapshot测试 1`] = `
<div
  class="home"
>
  <v-row-stub
    align="center"
    tag="div"
  >
    <v-col-stub
      class="text-center"
      tag="div"
    >
      <div
        class="my-2 "
      >
        <v-btn-stub
          activeclass=""
          color="error"
          depressed="true"
          large="true"
          tag="button"
          to="test"
          type="button"
        >
          点我跳转
        </v-btn-stub>
      </div>
    </v-col-stub>
  </v-row-stub>
   
  <helloworld-stub
    msg="Welcome to Your Vue.js App"
  />
</div>
`;
